### PR TITLE
Add Support for Agent Roles

### PR DESCRIPTION
**Pull request by AbhyudayaSharma**
This pull request is a draft because it depends on #3 ~and is a work-in-progress~.

### CODE REVIEW

**Comment by AbhyudayaSharma** [comment]
I think that this PR is ready now. Just waiting for a conclusion on #3

**Comment by AbhyudayaSharma** [comment]
CI not triggering a build

File: src/main/java/io/jenkins/plugins/folderauth/FolderBasedAuthorizationStrategy.java [hunk 1]
```diff
@@ -280,14 +377,33 @@ public void assignSidToFolderRole(String roleName, String sid) throws IOExceptio
...
+    /**
+     * Updates the ACL for the agent role
+     * <p>
+     * <b>Note: does not invalidate the cache</b>
+     * <p>
+     * Should be called when an agentRole has been updated.
+     *
+     * @param role the role to be updated
+     */
+    private void updateAclForAgentRole(@Nonnull AgentRole role) {
```
Comment (oleg-nenashev) [inline src/main/java/io/jenkins/plugins/folderauth/FolderBasedAuthorizationStrategy.java]: Not important: could look better with "ACL"

**Comment by AbhyudayaSharma** [comment]
I am merging it now. After that, I will update #3 with the changes for agent roles.

**Pull request merged**